services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/server/DeviceRegistryHttpServer.java

```diff
@@ -13,12 +13,39 @@
 
 package org.eclipse.hono.deviceregistry.server;
 
+import java.util.Objects;
+
 import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.notification.NoOpNotificationSender;
+import org.eclipse.hono.notification.NotificationSender;
 import org.eclipse.hono.service.http.HttpServiceBase;
 
+import io.vertx.core.Future;
+
 /**
  * Default REST server for Hono's example device registry.
  */
 public class DeviceRegistryHttpServer extends HttpServiceBase<ServiceConfigProperties> {
 
+    protected NotificationSender notificationSender = new NoOpNotificationSender();
+
+    /**
+     * Sets the client to publish notifications about changes on devices.
+     *
+     * @param notificationSender The client.
```

Comment:
It looks like you want to use the server to make sure that the single notification sender is being started/stopped as part of this server's lifecycle. If so, then IMHO we should document this fact here, i.e. that the sender being set here will have its start() and stop() methods invoked during startup/shutdown of the server.